Library/Homebrew/formula_creator.rb

```diff
@@ -10,10 +10,10 @@ class FormulaCreator
     attr_accessor :name
 
     sig {
-      params(name: T.nilable(String), version: T.nilable(String), tap: T.nilable(String), url: String,
+      params(url: String, name: T.nilable(String), version: T.nilable(String), tap: T.nilable(String),
              mode: T.nilable(Symbol), license: T.nilable(String), fetch: T::Boolean, head: T::Boolean).void
     }
-    def initialize(name, version, tap:, url:, mode:, license:, fetch:, head:)
+    def initialize(url:, name: nil, version: nil, tap: nil, mode: nil, license: nil, fetch: false, head: false)
```

Comment:
Don't set default arguments here unless they are needed by at least one caller.

Reply:
@abitrolly Feels like it might be nicer to just keep the original API then and not use keyword arguments?

Reply:
@abitrolly Sorry, I don't understand why this API is any better than what's there before. You'll need to try a bit harder to explain and convince me here.

Reply:
If default are removed back to original API, then in tests, instead of this:
```
fc = described_class.new(url: test[:url])
```
It will have to be this
```
        fc = described_class.new(
          nil,
          nil,
          tap:     nil,
          url:     test[:url],
          name:    nil,
          version: nil,
          tap:     nil,
          mode: nil,
          license: nil,
          fetch:   false,
          head:    false,
        )```

Reply:
@abitrolly I can see a complete test and other uses of `name_from_url` now, thanks. Given that, I think it makes a lot more sense for `name_from_url` to get folded into `parse_url`. It may also make sense to fold in `parse_url` into the constructor.

In general: methods that are only called once make sense to be combined unless:
- the combined method is very very very long (and RuboCop will tell you off for this)
- the individual methods are called separately

I don't think that is the case for either here.

Reply:
Makes sense. Folded `name_from_url` into `parse_url`. I would leave folding everything into the constructor for another PR. For now calling `parse_url` makes it explicit that no other logic is happening, except detecting missing values from URL.